### PR TITLE
cmd/snap: create snap user directory when running parallel installed snaps

### PR DESCRIPTION
**Pull request by bboozzoo**
When running snaps with instance key, create a user directory for the
non-instance-keyed snap. The directory must exist to be used as mount point for
remapping user data locations, `/home/joe/snap/foo_bar/*` -> `/home/joe/snap/foo/*`

### CODE REVIEW

File: cmd/snap/cmd_run.go [hunk 1]
```diff
@@ -303,9 +303,18 @@ func createUserDataDirs(info *snap.Info) error {
...
-	for _, d := range []string{userData, commonUserData} {
+	instanceUserData := info.UserDataDir(usr.HomeDir)
+	instanceCommonUserData := info.UserCommonDataDir(usr.HomeDir)
+	createDirs := []string{instanceUserData, instanceCommonUserData}
+	if info.InstanceKey != "" {
+		// parallel instance snaps get additional mapping in their mount
+		// namespace, namely /home/joe/snap/foo_bar ->
+		// /home/joe/snap/foo, make sure that the mount point exists and
+		// is owned by the user
+		snapUserDir := filepath.Join(usr.HomeDir, dirs.UserHomeSnapDir, info.SnapName())
```
Comment (zyga) [inline cmd/snap/cmd_run.go]: CC @chipaca

Comment (zyga) [inline cmd/snap/cmd_run.go]: Above we have the pattern of calling methods like info.SomeUserSpecificLocation(usr.HomeDir), is there a reason not to do it here as well?

Comment (mvo5) [reply cmd/snap/cmd_run.go]: Yeah, this feels a bit inconsistent at first look at least.

Comment (bboozzoo) [reply cmd/snap/cmd_run.go]: Let me add a helper then. IIRC there's one more place that may need an update as well.

Comment (bboozzoo) [reply cmd/snap/cmd_run.go]: This is done now.

**Pull request merged**